### PR TITLE
Use standard algorithms instead of raw loops

### PR DESCRIPTION
**Pull request by akva2**
Address some static analyzer noise.

### CODE REVIEW

**Comment by akva2** [comment]
jenkins build this please

File: opm/input/eclipse/Schedule/Schedule.cpp [hunk 1]
```diff
@@ -113,15 +113,12 @@
...
-        for (const auto& pattern : patterns)
-            if (name_match(pattern, name)) return true;
-
-        return false;
+    bool name_match_any(const std::unordered_set<std::string>& patterns,
+                        const std::string& name)
+    {
+        return std::find_if(patterns.begin(), patterns.end(),
+                            [&name](const auto& pattern)
+                            { return Opm::shmatch(pattern, name); }) != patterns.end();
```
Comment (bska) [inline opm/input/eclipse/Schedule/Schedule.cpp:121]: Maybe even `std::any_of()` here?

**Review by bska — COMMENTED** [review]
Much appreciated.  Just one tiny suggestion.

File: opm/input/eclipse/Schedule/Schedule.cpp (hunk 1 above)
Comment (akva2) [reply opm/input/eclipse/Schedule/Schedule.cpp]: yeah, doh.

**Comment by akva2** [comment]
jenkins build this please

**Pull request merged**